src/Combinatorics/Matroids/properties.jl

```diff
@@ -0,0 +1,1040 @@
+    r = M.pm_matroid.RANK
+    n = M.pm_matroid.N_ELEMENTS
+    #if M is uniform, apply Cor. 8.6.3 otherwise Thm. 8.6.4
+    if M.pm_matroid.N_BASES==binomial(n,r)
+        if n>=2r+2 
+            return r+1 
+        elseif n<=2r-2 
+            n-r+1
+        else
+            return inf
+        end
+    end
+    return minimum([vertical_connectivity(M),girth(M)])
+end
+
+@doc Markdown.doc"""
+    tutte_polynomial(M::Matroid)
+
+Return the Tutte polynomial of `M`. This is polynomial in the variables x and y with integral coefficients.
+See Section 15.3 in [Oxl11](@cite).
+
+# Example
+```jldoctest
+julia> tutte_polynomial(fano_matroid())
+x^3 + 4*x^2 + 7*x*y + 3*x + y^4 + 3*y^3 + 6*y^2 + 3*y
+
+```
+"""
+function tutte_polynomial(M::Matroid)
+    R, (x, y) = PolynomialRing(ZZ, ["x", "y"])
+    poly = M.pm_matroid.TUTTE_POLYNOMIAL
+    exp = Polymake.monomials_as_matrix(poly)
+    return R(Vector{Int}(Polymake.coefficients_as_vector(poly)),[[exp[i,1],exp[i,2]] for i in 1:size(exp)[1]])
+end
+
+@doc Markdown.doc"""
+    characteristic_polynomial(M::Matroid)
+
+Return the characteristic polynomial of `M`. This is polynomial in the variable q with integral coefficients.
+It is computed as an evaluation of the Tutte polynmomial.
+See Section 15.2 in [Oxl11](@cite).
+
+# Example
+```jldoctest
+julia> characteristic_polynomial(fano_matroid())
+q^3 - 7*q^2 + 14*q - 8
+
+```
+"""
+function characteristic_polynomial(M::Matroid)
+    R, q = PolynomialRing(ZZ, 'q')
+    return (-1)^M.pm_matroid.RANK*tutte_polynomial(M)(1-q,0)
+end
+charpoly(M::Matroid) = characteristic_polynomial(M)
+
+@doc Markdown.doc"""
+    reduced_characteristic_polynomial(M::Matroid)
+
+Return the reduced characteristic polynomial of `M`. This is the quotient of the characteristic polynomial by (q-1).
+See Section 15.2 in [Oxl11](@cite).
+
+# Example
+```jldoctest
+julia> reduced_characteristic_polynomial(fano_matroid())
+q^2 - 6*q + 8
+
+```
+"""
+function reduced_characteristic_polynomial(M::Matroid)
+    R, q = PolynomialRing(ZZ, 'q')
+    p = characteristic_polynomial(M)
+    c = Vector{Int}(undef,degree(p))
+    s = 0
+    for i in 1:degree(p)
+        s-= coeff(p,i-1)
+        c[i] = s
+    end
+    return R(c)
+end
+
+# This function compares two sets A and B in reverse lexicographic order.
+# It assumes that both sets are of the same length and ordered.
+# It returns true if A is less than B in this order
+function revlex_order(A::AbstractVector{Int64}, B::AbstractVector{Int64})
+    @assert length(A) == length(B)
+    r = length(A)
+    if A[end] < B[end]
+        return true
+    elseif A[end] < B[end]
+        return false
+    elseif r > 1
+        return revlex_order(view(A,1:(r-1)),view(B,1:(r-1)))
+    else
+        return false
+    end
+end
+
+#This functions computes the matrix of characteristic vectors of all r element subsets of [n] in revlex order.
+function revlex_bases_matrix(r::Int64,n::Int64)
+    all_bases = Oscar.Hecke.subsets(Vector(1:n),r)
```

Comment:
You really can write `subsets` everywhere *inside* Oscar.jl. Outside you can write `Oscar.subsets`. Perhaps we should export it? @fieker @thofma ?
```suggestion
    all_bases = subsets(Vector(1:n),r)
```